### DISCUSSION TITLE
Error with OctoApp Companion installation

### DISCUSSION THREAD

**Discussion by FredericHobby** [Guilouz/Creality-Helper-Script-Wiki#574] [Issues]
Hi,

Everything i needed went perfectly installed until installing OctoApp Companion (K1, 1.3.3.8) end up with an error.
I tryed some things around and could not add zlib, here is the log :

Running in K1 and K1 Max OS mode
Checking required system packages are installed...
Package python3 (3.11.7-1) installed in root is up to date.
Package python3-pip (23.3.1-1) installed in root is up to date.
Requirement already satisfied: virtualenv in /opt/lib/python3.11/site-packages (20.26.2)
Requirement already satisfied: distlib<1,>=0.3.7 in /opt/lib/python3.11/site-packages (fr                       om virtualenv) (0.3.8)
Requirement already satisfied: filelock<4,>=3.12.2 in /opt/lib/python3.11/site-packages (                       from virtualenv) (3.14.0)
Requirement already satisfied: platformdirs<5,>=3.9.1 in /opt/lib/python3.11/site-package                       s (from virtualenv) (4.2.2)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting                        behaviour with the system package manager. It is recommended to use a virtual environment                        instead: https://pip.pypa.io/warnings/venv
System package install complete.
Checking Python Virtual Environment For OctoApp...
Updating PIP if needed... (this can take a few seconds or so)
Requirement already satisfied: pip in /usr/data/octoapp-env/lib/python3.11/site-packages (24.0)
Installing or updating required python libs...
  error: subprocess-exited-with-error

  × Building wheel for pillow (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [216 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-mips-cpython-311
      creating build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PdfImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/MpoImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/TarIO.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PixarImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ContainerIO.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageStat.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageQt.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/WebPImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/SunImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/TiffImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/FitsImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/BlpImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageShow.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageColor.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/CurImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PyAccess.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/EpsImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/Image.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/SpiderImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageSequence.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageMorph.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PdfParser.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageWin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PcfFontFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_tkinter_finder.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_deprecate.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/GdImageFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/features.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/McIdasImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/XpmImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PcxImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/GimpPaletteFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageTransform.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_version.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PcdImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_util.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/MicImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/IcoImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageGrab.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/WalImageFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/QoiImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageOps.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/BmpImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/IptcImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/FtexImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/MpegImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/FontFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/SgiImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ExifTags.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PpmImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/BdfFontFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/Jpeg2KImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/MspImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/GimpGradientFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/WmfImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageDraw.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/XVThumbImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PsdImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PalmImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageFont.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageMath.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/TgaImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/JpegImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageMode.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/IcnsImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/DcxImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/__init__.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageTk.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageCms.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/XbmImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/GbrImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImtImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/FliImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageChops.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/GribStubImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/DdsImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_typing.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PaletteFile.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/TiffTags.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/report.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/FpxImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/BufrStubImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/__main__.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/JpegPresets.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageFilter.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/Hdf5StubImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PngImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/PSDraw.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageDraw2.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImageEnhance.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImagePalette.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/ImagePath.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_binary.py -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/GifImagePlugin.py -> build/lib.linux-mips-cpython-311/PIL
      running egg_info
      writing src/pillow.egg-info/PKG-INFO
      writing dependency_links to src/pillow.egg-info/dependency_links.txt
      writing requirements to src/pillow.egg-info/requires.txt
      writing top-level names to src/pillow.egg-info/top_level.txt
      reading manifest file 'src/pillow.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.c'
      warning: no files found matching '*.h'
      warning: no files found matching '*.sh'
      warning: no files found matching '*.txt'
      warning: no files found matching '.flake8'
      warning: no previously-included files found matching '.appveyor.yml'
      warning: no previously-included files found matching '.clang-format'
      warning: no previously-included files found matching '.coveragerc'
      warning: no previously-included files found matching '.editorconfig'
      warning: no previously-included files found matching '.readthedocs.yml'
      warning: no previously-included files found matching 'codecov.yml'
      warning: no previously-included files found matching 'renovate.json'
      warning: no previously-included files matching '.git*' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      no previously-included directories found matching '.ci'
      no previously-included directories found matching 'wheels'
      adding license file 'LICENSE'
      writing manifest file 'src/pillow.egg-info/SOURCES.txt'
      copying src/PIL/_imaging.pyi -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_imagingcms.pyi -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_imagingft.pyi -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_imagingmath.pyi -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_imagingmorph.pyi -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/_webp.pyi -> build/lib.linux-mips-cpython-311/PIL
      copying src/PIL/py.typed -> build/lib.linux-mips-cpython-311/PIL
      running build_ext

      The headers or library files could not be found for zlib,
      a required dependency when compiling Pillow from source.

      Please see the install instructions at:
         https://pillow.readthedocs.io/en/latest/installation.html

      Traceback (most recent call last):
        File "<string>", line 1009, in <module>
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 104, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 184, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
          dist.run_commands()
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 967, in run_command
          super().run_command(command)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/normal/lib/python3.11/site-packages/wheel/bdist_wheel.py", line 368, in run
          self.run_command("build")
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 967, in run_command
          super().run_command(command)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 132, in run
          self.run_command(cmd_name)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 967, in run_command
          super().run_command(command)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 91, in run
          _build_ext.run(self)
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
          self.build_extensions()
        File "<string>", line 826, in build_extensions
      RequiredDependencyException: zlib

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/usr/data/octoapp-env/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/data/octoapp-env/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/data/octoapp-env/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/data/opt/tmp/pip-install-8dvqrbx_/pillow_27191b65cdbe469092420b785914f313/_custom_build/backend.py", line 26, in build_wheel
          return super().build_wheel(wheel_directory, config_settings, metadata_directory)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 410, in build_wheel
          return self._build_with_temp_dir(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 395, in _build_with_temp_dir
          self.run_setup()
        File "/usr/data/opt/tmp/pip-install-8dvqrbx_/pillow_27191b65cdbe469092420b785914f313/_custom_build/backend.py", line 20, in run_setup
          return super().run_setup(setup_script)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/data/opt/tmp/pip-build-env-hpdimd55/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 1025, in <module>
      RequiredDependencyException:

      The headers or library files could not be found for zlib,
      a required dependency when compiling Pillow from source.

      Please see the install instructions at:
         https://pillow.readthedocs.io/en/latest/installation.html

      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pillow
ERROR: Could not build wheels for pillow, which is required to install pyproject.toml-based projects

----End of log---

**Comment by mindkill666** [comment]
Having the same problem, i believe there's a problem with other companion apps too, probably something to do with python upgrade.
I think they are working on it.

**Comment by Guilouz** [comment]
Fixed issue on OctoApp repo : https://github.com/crysxd/OctoApp-Plugin/pull/78

Needed to wait is merged.

**Comment by mindkill666** [comment]
Still getting this error:

 note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pillow
ERROR: Could not build wheels for pillow, which is required to install pyproject.toml-based projects

**Comment by Guilouz** [comment]
@mindkill666 just installed on a fresh install after factory reset and it's working fine.

Issue is on your side :

<img width="1720" alt="Capture d’écran 2024-05-19 à 22 32 17" src="https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/12702322/cd87cc61-9a62-4d3d-9a1b-b6f8c8a4d4fa">
